### ISSUE TITLE
FILENAME_MISSING_COMPONENTS is based on advice, not a rule

### ISSUE THREAD

**Issue by paulehoffman** [ietf-tools/idnits#32]
`draft-rswg-rfc7997bis` is draft in the RSWG, and the filename is completely legal. The description of "Filename must consists of at least 4 components" does not match some of the names described on https://authors.ietf.org/naming-your-internet-draft. 

This file also gets FILENAME_INVALID_VERSION_SUFFIX and FILENAME_DOCNAME_MISMATCH errors, which is probably queued off of the above. The docname in the file is `draft-rswg-rfc7997bis-02`, which has a valid version suffix.

**Comment by markusa** [comment]
Same problem here. I want to use idnits with [github workflows](https://github.com/tsvwg/ietf-multipath-dccp/blob/master/.github/workflows/verify.yml#L34-L35) to integrate it into the automatic build process from Markdown to XML/TXT/HTML.

However, idnits v3 complains about the filename of `draft-ietf-tsvwg-multipath-dccp.txt`, which is the input file, and refuses further action. If this is desired, it would at least be good to have this strict file name check configurable.

```
 FAIL  Document is INVALID. ❌

 1  Error
 └- Code - FILENAME_INVALID_VERSION_SUFFIX
 └- Desc - Filename must end with a version in format 00.
 └- Ref  - https://authors.ietf.org/naming-your-internet-draft

 2  Error
 └- Code - FILENAME_DOCNAME_MISMATCH
 └- Desc - Filename does not match the name declared in the document.
 └- Ref  - https://authors.ietf.org/naming-your-internet-draft
```


A run of `https://raw.githubusercontent.com/tsvwg/ietf-multipath-dccp/master/draft-ietf-tsvwg-multipath-dccp.mkd` with https://author-tools.ietf.org/idnits, which actually uses idnits 2.17.1, only throws a warning, which is sufficient in my opinion.

```
  == Mismatching filename: the document gives the document name as
     'draft-ietf-tsvwg-multipath-dccp-15', but the file name used is
     'draft-ietf-tsvwg-multipath-dccp'
```

**Comment by rjsparks** [comment]
@1456055067 proposed a fix at #262 that would allow `draft-ietf-01.txt` to be submitted.

This is not right - a document with that name should never enter the repository.

@paulehoffman 's request is that idnits not force `draft-editorial-rswg-subjectofinterest-{(version_number}} - the complaint was the check enforcing that the stream name appear in the second tuple if the document declared it was in a stream.

@markusa 's request is different - that request is to have a mode where idnits can run over a document that has a name that will fail submission checks (in that it doesn't have a version). Providing that might be useful, but I think it would need to be a different MODE and the behavior of that mode would need to be carefully specified (@martinthomson may have input).

There is also tension between looking at the name of the file being fed to the tool and what the document declares its name is. This should be considered separately in the txt and xml checkers. The xml checker probably shouldn't _care_ what the input filename is at all (at most it might warn if they are different).

**Comment by paulehoffman** [comment]
And just to add an additional headache that I think might be related: https://datatracker.ietf.org/doc/draft-ietf-deleg/ Yes, that's both the WG name and the document name (because it is the protocol name). Should be headed to IETF Last Call within a few monthsl

**Comment by 1456055067** [comment]
When I first looked at this, I saw that the 4-component requirement was more of an advisory, not a strict rule. Some valid
draft names like "draft-rswg-rfc7997bis" have only 3 components, just like the draft-ietf-deleg that @paulehoffman highlighted above. @rjsparks I do agree that something like draft-ietf-01 should not be allowed.

I was thinking about a regex with a negative lookahead to prevent  somthing like draft-ietf-01 from being allowed (e.g., ^draft-([a-zA-Z]+)-(?!\d)[a-zA-Z]+.+$)

**Comment by paulehoffman** [comment]
I think you misunderstood my comment about draft-ietf-deleg. In my mind, idnits and/or the Datatracker should have seen that the second label was "ietf" and disallowed a three-label filename immediately.

**Comment by 1456055067** [comment]
Indeed, I did misunderstand this. I went and read [naming-your-internet-draft ](https://authors.ietf.org/naming-your-internet-draft). Even under formats, I see in the section that they call out:
- draft-(source)-(group)-(subject)-(version)
- draft-(source)-(subject)-(version)

Both draft-rswg-rfc7997bis and draft-ietf-deleg meet the version addition requirement. Are we suggesting that this should not be the case? If so, what should the recommendation be to prevent this in the future? Do we want to force the (group) section upon a I-D being adopted by one of the streams outlined in [naming-your-internet-draft ](https://authors.ietf.org/naming-your-internet-draft#source)?

**Comment by martinthomson** [comment]
> that request is to have a mode where idnits can run over a document that has a name that will fail submission checks 

Just on this part, since I was invoked.  I had imagined that this tool would ultimately settle on a set of errors, where each is clearly identified (the FILENAME_ labels fit that) and independently available for suppression.  In that case, you might choose to suppress the FILENAME_INVALID_VERSION_SUFFIX error in cases where no version is present.  I don't believe that I have a need for that; the toolchain I maintain invokes idnits with a versioned file.  But the general principle is useful outside of this narrow case.
